azure-mgmt-resources/src/main/java/com/microsoft/azure/management/resources/fluentcore/utils/RXMapper.java

```diff
@@ -0,0 +1,48 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+
+package com.microsoft.azure.management.resources.fluentcore.utils;
+
+import rx.Observable;
+import rx.functions.Func1;
+import rx.schedulers.Schedulers;
+
+/**
+ * An internal utility class representing an RX function returning the provided type instance
+ * from a call with an arbitrary parameter.
+ * @param <T> the type to emit as Observable
+ */
+public final class RXMapper<T> implements Func1<Object, T> {
+    private final T value;
+
+    /**
+     * Shortcut for mapping an arbitrary observable to one returning an instance of a specific type, using the IO scheduler.
+     * @param fromObservable an observable
+     * @param toValue the value to emit to the observer
+     * @param <T> the type of the value to emit
+     * @return an observable to be emitted
+     */
+    public static <T> Observable<T> map(Observable<?> fromObservable, final T toValue) {
+        if (fromObservable != null) {
+            return fromObservable.subscribeOn(Schedulers.io())
+                    .map(new RXMapper<T>(toValue));
+        } else {
+            return null;
```

Comment:
`return Observable.empty()` or `return Observable.just(toValue)` if we want to return `toValue` as default value as well when fromObservable is null.